internal/osutils/lockfile/pidlock.go

```diff
@@ -94,6 +96,28 @@ func (pl *PidLock) Close(keepFile ...bool) error {
 	return nil
 }
 
+// WaitForLock will attempt to acquire the lock for the duration given
+func (pl *PidLock) WaitForLock(timeout time.Duration) error {
+	lockedErr := &AlreadyLockedError{}
+	timer := time.NewTimer(timeout)
+	for {
+		_, err := pl.TryLock()
+		if err != nil {
+			if !errors.As(err, &lockedErr) {
```

Comment:
```suggestion
			if !errors.As(err, &lockedErr) {
```

Use `errs.Matches(err, &AlreadyLockedError{})`.